web/source/text/outputTarget.ts

```diff
@@ -44,6 +82,84 @@ namespace com.keyman.text {
       this._dks = dks.clone();
     }
 
+    /**
+     * Determines the basic operations needed to reconstruct the current OutputTarget's text from the prior state specified
+     * by another OutputTarget based on their text and caret positions.
+     * 
+     * This is designed for use as a "before and after" comparison to determine the effect of a single keyboard rule at a time.
+     * As such, it assumes that the caret is immediately after any inserted text.
+     * @param from An output target (preferably a Mock) representing the prior state of the input/output system.
+     */
+    private buildTransformFrom(original: Mock): Transform {
+      let to = this.getText();
+      let from = original.getText();
+
+      let fromCaret = original.getDeadkeyCaret();
+      let toCaret = this.getDeadkeyCaret();
+
+      // Step 1:  Determine the number of left-deletions.
+      for(var newCaret=0; newCaret < fromCaret; newCaret++) {
```

Comment:
Should probably use `let` not `var`?

Reply:
I suppose so.  I thought `let` forced the variable to be `const`, but that's Swift, not TypeScript.  The joys of working in multiple programming languages...

That said, in case our iOS dev(s) ever need to maintain the code, keeping it as `var` will be one less point of confusion for them.

Reply:
I'm not keen on cross-language decision pollution :) 

To help our poor Swift devs, we should use `const` where we can, and `let` elsewhere. As far as possible we should avoid use of `var` per https://www.typescriptlang.org/docs/handbook/variable-declarations.html